src/elements/Chip/Chip.tsx

```diff
@@ -70,17 +83,55 @@ export const Chip: FC<ChipProps & ChipRef> = forwardRef(
         [css[color]]: true,
         [css[variant]]: true,
         [css[size]]: true,
-        [css.selectable]: onClick && !disabled,
         [css.selected]: !!selected,
         [css.disabled]: disabled,
+        [css.selectable]: onClick && !disabled,
+        [css.canDelete]: !!onDelete,
         [css.disableMargins]: disableMargins
       })}
     >
-      {children}
+      <div className={css.contents}>
+        {icon && icon}
+        <span className={css.label}>{children}</span>
+        {onDelete && (
+          <Button
+            tabIndex={0}
+            variant="text"
+            className={css.delete}
+            onClick={event => {
+              if (!disabled) {
+                event.stopPropagation();
+                onDelete();
+              }
+            }}
+            disabled={disabled}
+            disableMargins
+            disablePadding
+          >
+            <CloseIcon />
+          </Button>
+        )}
+      </div>
     </div>
   )
 );
 
+const CloseIcon = () => (
```

Comment:
Let's move this into another file. Also make it a prop so users can override it.

Reply:
actually i was going to ask you about this.. it is in another file but i couldn't get it to import correctly

Reply:
potentially, you had the same issue - looking at this:
https://github.com/reaviz/reablocks/blob/master/src/form/Input/Input.story.tsx#L21